BridgeCareApp/BridgeCareCodeFirst/Services/DetailedReportRepository.cs

```diff
@@ -0,0 +1,294 @@
+﻿using BridgeCare.Interfaces;
+using BridgeCare.Models;
+using OfficeOpenXml;
+using OfficeOpenXml.Style;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+
+namespace BridgeCare.Services
+{
+    public class DetailedReportRepository : IReportCreater
+    {
+        private Dictionary<bool, Action<ConditionalData, ExcelWorksheet>> ExcelValues = new Dictionary<bool, Action<ConditionalData, ExcelWorksheet>>();
+        private readonly BridgeCareContext db;
+        private readonly IDetailedReport detailedReport;
+        private readonly IDeficient deficientResult;
+        private readonly ITarget targets;
+
+        private readonly IBudget budgetReport;
+        public readonly FillWorkSheet fillWorkSheet;
+        public DetailedReportRepository(IBudget report, IDetailedReport yearlyReport,
```

Comment:
This parameter list is getting large. Seek a way to compose abstractions together to reduce this issue.